### PR TITLE
SpringBoot HW1 - set (1) basic required SpringBoot functionality and (2) implement functionality to read from CSV file

### PR DESCRIPTION
**Pull request by fyarkiy**

### CODE REVIEW

**Review by jah701 — APPROVED** [review]
Good job! :)

**Comment by RomanDubovskyi** [comment]
write informative PR names 
image

File: empty_file.csv [hunk 1]
```diff
@@ -0,0 +1 @@
+1,B001E4KFG0,A3SGXH7AUHU8GW,delmartian,1,1,5,1303862400,Good Quality Dog Food,I have bought several of the Vitality canned dog food products and have found them all to be of good quality. The product looks more like a stew than a processed meat and it smells better. My Labrador is finicky and she appreciates this product better than  most.
```
Comment (RomanDubovskyi) [inline empty_file.csv]: empty line in the end of the file

File: incorrect_data.csv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Id,ProductId,UserId,ProfileName,HelpfulnessNumerator,HelpfulnessDenominator,Score,Time,Summary,Text
+AAA,B001E4KFG0,A3SGXH7AUHU8GW,delmartian,1,1,5,1303862400,Good Quality Dog Food,I have bought several of the Vitality canned dog food products and have found them all to be of good quality. The product looks more like a stew than a processed meat and it smells better. My Labrador is finicky and she appreciates this product better than  most.
```
Comment (RomanDubovskyi) [inline incorrect_data.csv]: same

File: three_records.csv [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+2,B00813GRG4,A1D87F6ZCVE5NK,dll pa,0,0,1,1346976000,Not as Advertised,"Product arrived labeled as Jumbo Salted Peanuts...the peanuts were actually small sized unsalted. Not sure if this was an error or if the vendor intended to represent the product as ""Jumbo""."
+3,B000LQOCH0,ABXLMWJIXXAIN,"Natalia Corres ""Natalia Corres""",1,1,4,1219017600,"""Delight"" says it all","This is a confection that has been around a few centuries.  It is a light, pillowy citrus gelatin with nuts - in this case Filberts. And it is cut into tiny squares and then liberally coated with powdered sugar.  And it is a tiny mouthful of heaven.  Not too chewy, and very flavorful.  I highly recommend this yummy treat.  If you are familiar with the story of C.S. Lewis' ""The Lion, The Witch, and The Wardrobe"" - this is the treat that seduces Edmund into selling out his Brother and Sisters to the Witch."
```
Comment (RomanDubovskyi) [inline three_records.csv]: and here

File: src/test/java/ma/boot/springboot/ReadingCsvFileTests.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+public class ReadingCsvFileTests {
+    private static final String CORRECT_CSV_FILE = "three_records.csv";
+    private static final String NOT_EXISTED_FILE = "file123.csv";
+    private static final String WRONG_DATA = "incorrect_data.csv";
+    private static final String EMPTY_FILE = "empty_file.csv";
+
+    @Test
+    public void loadCorrectFile() throws Exception {
+        int result = ReadingCsvFile.readWithCsvBeanReader(CORRECT_CSV_FILE);
```
Comment (RomanDubovskyi) [inline src/test/java/ma/boot/springboot/ReadingCsvFileTests.java]: what is result and 3 - it's not clear for me in this code

File: src/main/java/ma/boot/springboot/service/read/ReadingCsvFile.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import org.supercsv.exception.SuperCsvException;
+import org.supercsv.io.CsvBeanReader;
+import org.supercsv.io.ICsvBeanReader;
+import org.supercsv.prefs.CsvPreference;
+import org.supercsv.util.CsvContext;
+
+public class ReadingCsvFile {
+    private static final Logger logger = Logger.getLogger(ReadingCsvFile.class);
+
+    public static int readWithCsvBeanReader(String fileName) throws IOException {
```
Comment (RomanDubovskyi) [inline src/main/java/ma/boot/springboot/service/read/ReadingCsvFile.java]: this method should return list of reviews. 
And if you need number of records just call reviews.size()

File: src/main/java/ma/boot/springboot/service/read/ReadingCsvFile.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        return records;
+    }
+
+    private static CellProcessor[] getProcessors() {
+        final CellProcessor parseDate = new CellProcessorAdaptor() {
+            @Override
+            public Object execute(Object o, CsvContext csvContext) {
+                Date date = new Date(Long.parseLong(o.toString()) * 1000L);
+                SimpleDateFormat jdf = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
```
Comment (RomanDubovskyi) [inline src/main/java/ma/boot/springboot/service/read/ReadingCsvFile.java]: declare format as a constant and improve naming

File: src/test/java/ma/boot/springboot/ReadingCsvFileTests.java (hunk 4 above)
Comment (fyarkiy) [reply src/test/java/ma/boot/springboot/ReadingCsvFileTests.java]: Number of records which were read. I have already changed this in SB2 - I am creating a list with read records

File: src/main/java/ma/boot/springboot/service/read/ReadingCsvFile.java (hunk 5 above)
Comment (fyarkiy) [reply src/main/java/ma/boot/springboot/service/read/ReadingCsvFile.java]: Realised this in SB-2 and changed it there. Will do it here as well

File: src/test/java/ma/boot/springboot/ReadingCsvFileTests.java [hunk 7]
```diff
@@ -10,16 +10,17 @@
     private static final String NOT_EXISTED_FILE = "file123.csv";
     private static final String WRONG_DATA = "incorrect_data.csv";
     private static final String EMPTY_FILE = "empty_file.csv";
+    private static final int NUMBER_OF_RECORDS = 3;
 
     @Test
     public void loadCorrectFile() throws Exception {
-        int result = ReadingCsvFile.readWithCsvBeanReader(CORRECT_CSV_FILE);
-        Assert.assertEquals(3, result);
+        int result = ReadingCsvFile.readWithCsvBeanReader(CORRECT_CSV_FILE).size();
```
Comment (RomanDubovskyi) [inline src/test/java/ma/boot/springboot/ReadingCsvFileTests.java:17]: you may use name recordsCount instead, it will be more informative

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**